Network/HTTP/ReverseProxy.hs

```diff
@@ -131,6 +132,33 @@ rawProxyTo getDest appdata = do
         fromServer = DCN.appSource appdataServer
         toServer = DCN.appSink appdataServer
 
+-- | Set up a reverse tcp proxy server, which will have a minimal overhead.
+--
+-- This function uses raw sockets, parsing as little of the request as
+-- possible. The workflow is:
+--
+-- 1. Open up a connection to the given host\/port.
+--
+-- 2. Pass all bytes across the wire unchanged.
+--
+-- If you need more control, such as modifying the request or response, use 'waiProxyTo'.
+rawTcpProxyTo :: (MonadBaseControl IO m, MonadIO m)
+           => ProxyDest
+           -> AppData
+           -> m ()
+rawTcpProxyTo (ProxyDest host port) appdata = liftIO $
+    DCN.runTCPClient (DCN.clientSettings port host) withServer
+  where
+    withServer appdataServer = do
```

Comment:
I'd recommend implementing this with `race` from `async` instead. That said: is there a good reason for these semantics? It seems like it would make more sense to have a `concurrently` semantics, allowing both ends of the `Socket` to exit before closing the proxy.